build/buildHelpers.js

```diff
@@ -0,0 +1,202 @@
+    logToTerminal(`Starting to build ${bundleName}`);
+    config.watch = watch;
+    config.bail = true; // Causes webpack to break upon first encountered error. Pretty annoying when build errors scroll off the screen.
+    let startTime = process.hrtime();
+    webpack(config, (err, stats) => {
+        if (!err) {
+            let msg = `Finished building ${bundleName}`;
+            //first run, add nice timer.
+            if (callback) {
+                let end = process.hrtime(startTime);
+                msg += ` after ${chalk.magenta(prettyHrtime(end))}`;
+            }
+            logToTerminal(msg, "cyan");
+        } else {
+            console.error(errorOutColor("Webpack Error.", err));
+        }
+        if (stats.hasErrors()) {
+            console.error(errorOutColor(stats.toJson().errors));
+        }
+        // Webpack will call this function every time the bundle is built.
+        // Webpack is run in "watch" mode which means this function will be called over and over and over.
+        // We only want to invoke the async callback back to the gulp file once - the initial webpack build.
+        if (callback) {
+            callback();
+            callback = undefined;
+        }
+    });
+}
+
+/**
+ * Returns the value for the given name, looking in (1) environment variables, (2) command line args
+ * and (3) startupConfig. For instance, `set BLAH_BLAH=electron` or `npx gulp dev --blah_blah:electron`
+ * This will search for both all caps, all lowercase and camelcase.
+ * @param {string} name The name to look for in env variables and args
+ * @param {string} defaultValue The default value to return if the name isn't found as an env variable or arg
+ */
+const envOrArg = (name, defaultValue) => {
+    let lc = name.toLowerCase();
+    let uc = name.toUpperCase();
+    let cc = name.replace(/(-|_)([a-z])/g, function (g) { return g[1].toUpperCase(); });
+
+    // Check environment variables
+    if (env[lc]) return env[lc];
+    if (env[uc]) return env[uc];
+
+    // Check command line arguments
+    lc = "--" + lc + ":";
+    uc = "--" + uc + ":";
+    let rc = null;
+    process.argv.forEach(arg => {
+        if (arg.startsWith(lc)) rc = arg.split(lc)[1];
+        if (arg.startsWith(uc)) rc = arg.split(uc)[1];
+    });
+
+    // Look in startupConfig
+    if (!rc) {
+        rc = startupConfig[env.NODE_ENV][cc] || startupConfig[env.NODE_ENV][lc] || startupConfig[env.NODE_ENV][uc];
+    }
+    rc = rc || defaultValue;
+    return rc;
+}
+
+/**
+ * Given an array of webpack parallel config objects, this function will spool up a maximum
+ * of N workers (where N is limited by CPU cores). When all of them finish building, the done
+ * callback will be invoked.
+ * 
+ * @param {WebpackParallelConfig[]} WebpackParallelConfigs An array of objects that describe a webpack config
+ * @param {Function} done 
+ */
+const runWebpackInParrallel = (WebpackParallelConfigs, exitOnCompletion, done) => {
```

Comment:
```suggestion
const runWebpackInParrallel = (webpackParallelConfigs, exitOnCompletion, done) => {
```
The UpperCamelCase makes me think it's some sort of type or class or something.

Reply:
Mistake on my part using ctrl+d. good catch